CGPs/cgp-0053.md

```diff
@@ -0,0 +1,100 @@
+---
+cgp: <to be assigned>
+title: Increase the gasLimit to 50M and gas targetDensity to 0.7
+date-created: 2022-04-03
+author: Mariano Cortesi (@mcortesi5), Javier Cortejoso (@jcortejoso), Gastón Ponti (@gastonponti)
+status: DRAFT
+discussions-to: <tbd>
+governance-proposal-id: <tbd>
+date-executed: <tbd>
+---
+## Overview
+
+CGP - Celo Governance Proposal
+
+This proposal sets the block gas limit parameter of the Celo blockchain from `20M` to `50M` and the gasPriceMinimum's targetDensity from `0.8` to `0.7`. The change is motivated by the improvement on block processing throughput with the latest celo-blockchain releases.
+
+The most relevant changes that positively impact the performance of the network are:
+
+- [Snapshots](https://blog.ethereum.org/2021/03/03/geth-v1-10-0/)
+- [CIP43: Block Context](https://github.com/celo-org/celo-proposals/blob/master/CIPs/cip-0043.md)
+
+For confirming the safety and stability of a block gas limit of `50M`, cLabs Blockchain Team have run load and stability testing during the last weeks.
+
+The change will increase the transaction throughput by way of the increasing the block space by 150%, and also will increase the effective network target block for adjusting the baseFee, that currently is `0.8 * blockSize` (`16M`) and will be `0.7 * blockSize` (`35M`).
+
+## Proposed Changes
+
+1. Increase the block gas limit parameter:
+
+    - Destination: Call `setBlockGasLimit` on the `BlockchainParameters` smart contract owned by `Governance`
+    - Data: 50 000 000 (50 Million)
+    - Value: 0 (NA)
+
+2. Decrease the gasPriceMinimum's target density:
+
+    - Destination: Call `setTargetDensity` on the `GasPriceMinimum` smart contract owned by `Governance`
+    - Data: 700 000 000 000 000 000 000 000 (7e23 => 0.7)
```

Comment:
It would be nice to have a bit of an explanation of this value, which would allow people to independently verify it. Its not clear why 7e23 would correspond to 0.7, so its difficult to check that its not off by a factor of 10/100/1000 ... etc

Reply:
Not a big expert for explaining the rationale about this format, but the value is used by `[setTargetDensity](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/common/GasPriceMinimum.sol#L92)` call, that `[wraps](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/common/FixidityLib.sol#L46)` the value as a `[Fraction](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/common/FixidityLib.sol#L18)` struct:

```
  /**
   * @notice Wrap a uint256 that represents a 24-decimal fraction in a Fraction
   * struct.
   * @param x Number that already represents a 24-decimal fraction.
   * @return A Fraction struct with contents x.
   */
  function wrap(uint256 x) internal pure returns (Fraction memory) {
    return Fraction(x);
  }
  ```